### ISSUE TITLE
Default PWM frequency is very low, resulting flickering and noisy motor control

### ISSUE THREAD

**Issue by gomakekit** [microsoft/pxt-microbit#5446]
**Describe the bug**
The default PWM frequency on the pin outputs is very low, resulting flickering and noisy motor operation when using analog write.

**To Reproduce**
In makecode: 
basic.forever(function () {
    pins.analogWritePin(AnalogPin.P0, 300)
})

**Expected behavior**
Led flickering can be seen with the bare eye. You can also see it as a "trail" when the led moves.

When using motor control via a FET transistor, the operation will become very noisy. Video: https://youtu.be/Fer2_VP-BQo

![Screenshot 2024-01-30 at 17 01 11](https://github.com/microsoft/pxt-microbit/assets/56154537/c0606b01-1a9b-4d20-82d2-d0173cdaed7d)

**micro:bit version (please complete the following information):**
V1 and V2

**Additional context**
Why not make default PWM 200 instead of 20000?

I have tested the following code with a cheap servo to see if the servo will get affected after changing the pwm frequency, and the servo still works like before. 

pins.analogWritePin(AnalogPin.P1, 0)
pins.analogSetPeriod(AnalogPin.P1, 200)
basic.forever(function () {
    pins.servoWritePin(AnalogPin.P1, 45)
    basic.pause(1000)
    pins.servoWritePin(AnalogPin.P1, 135)
    basic.pause(1000)
})

**Comment by abchatra** [comment]
@carlosperate  any thoughts in this bug? Should we change the default for this?

**Comment by abchatra** [comment]
@carlosperate ?

**Comment by microbit-carlos** [comment]
The 20ms period was likely picked to match the period of servos by default. 

The servo functions does set the period to 20ms as well, so code like this `pins.analogSetPeriod(AnalogPin.P1, 200); pins.servoWritePin(AnalogPin.P1, 45)` would have changed the period to 20 ms automatically inside `servoWritePin()`.

My concern to change the default period would be that perhaps some projects or extensions might be depending on that period being the default in their PWM output. For finer motor control the period can always be set to something smaller, either by an extension or provided instructions.

@gomakekit I assume the main advantage of changing the default might be to have smaller programmes for the user? Or is there any other reason this might be useful? Does it clash with other analog write functionality?

**Comment by gomakekit** [comment]
So the reason we are so found of analog write is that instead of making a library for each product, is that user can simply use the analog write to control the motor speed or led brightness in one block (typically bubble machine motor speed). However the set period isnt that easy to grasp, also there is the drawback having to make an analog write before you can set the period, hard to explain to users. We (MakeKit) are teaching the universal function of analog write instead of "hovercraft speed" or similar. The only possible drawback for a pwm frequency is limited to transistor timing and recovery. I have found that the cheapest low power (fet) transistors can handle at least 1 mhz, typically much more. With a timing of 200 microseconds, we are only at 5000 hz, so we are on the safe side. Another way of looking at this is that old led dimmers flicker, while the new ones don't. Please consider changing the default period. Thanks!

**Comment by gomakekit** [comment]
I would say the most natural solution would be that a servo write would assign that frequency to 20ms, while an analog write would assign it to 10khz automatically. As far as I understood, the pins should be independently able to set frequencies.

**Comment by fabianhugo** [comment]
I agree, that would be a good solution. I don't know if you refer to the general hardware capabilities or the codal, but in the codal it is currently not possible to set pin frequencies independently, try e.g. the following code and probe with an oscilloscope: 
https://makecode.microbit.org/_egW04T7Am8kf
There the servo write function in fact sets the PWM frequency back to 50Hz, but that affects the other PWM frequencies as well, so the frequency of P0 goes from 10kHz to 50Hz. Interestingly, also the duty cycle changes from 50% to 96%. This [issue](https://github.com/microsoft/pxt-microbit/issues/4950) contains more information on that.
